### PR TITLE
feat: use shadow dom to attach context vue

### PR DESCRIPTION
**Pull request by tshemeng**
feat: use shadow dom to attach context vue

### CODE REVIEW

File: src/contentScripts/index.ts [hunk 1]
```diff
@@ -12,5 +13,12 @@ onMessage('tab-prev', ({ data }) => {
 
 // mount component to context window
 const container = document.createElement('div')
+const shadowDOM = container.attachShadow?.({ mode: 'closed' }) || container
```
Comment (tmkx) [inline src/contentScripts/index.ts:16]: In consideration of debug experience, could we use `open` in dev ?

Comment (tshemeng) [reply src/contentScripts/index.ts]: updated.

File: src/contentScripts/index.ts [hunk 2]
```diff
@@ -1,8 +1,11 @@
 /* eslint-disable no-console */
 import { onMessage } from 'webext-bridge'
+import browser from 'webextension-polyfill'
 import { createApp } from 'vue'
 import App from './Content.vue'
 
+const isDev = browser.runtime.getManifest()?.content_security_policy?.includes('http://localhost') === true
```
Comment (tmkx) [inline src/contentScripts/index.ts:7]: So tricky, why not try to [define](https://vitejs.dev/config/#define) a `__DEV__` constant? and it's even tree-shakable :smile:

Comment (tshemeng) [reply src/contentScripts/index.ts]: updated. thanks, I learned a lot from your review.

File: src/contentScripts/index.ts [hunk 3]
```diff
@@ -1,16 +1,29 @@
 /* eslint-disable no-console */
 import { onMessage } from 'webext-bridge'
+import browser from 'webextension-polyfill'
 import { createApp } from 'vue'
 import App from './Content.vue'
 
+declare const __DEV__: Boolean
```
Comment (tmkx) [inline src/contentScripts/index.ts:7]: using `boolean` instead of `Boolean`

File: src/contentScripts/index.ts [hunk 4]
```diff
@@ -1,16 +1,29 @@
...
+
 console.info('[vitesse-webext] Hello world from content script')
 
 // communication example: send previous tab title from background page
 onMessage('tab-prev', ({ data }) => {
   console.log(`[vitesse-webext] Navigate from page "${data.title}"`)
 })
 
+console.log(__DEV__)
+console.log(typeof __DEV__)
```
Comment (tmkx) [inline src/contentScripts/index.ts:17]: remove test code

**Pull request merged**